sdk/platform/http_client/curl/src/az_curl.c

```diff
@@ -170,6 +170,45 @@ static AZ_NODISCARD az_result _az_http_client_curl_add_header_to_curl_list(
   return result;
 }
 
+/**
+ * @brief Adds special header "Expect:" for libcurl to avoid sending only headers to server and wait
+ * for a 100 Continue response before sending a PUT method
+ *
+ * see: https://github.com/curl/curl/blob/master/docs/FAQ#L1033
+ * libcurl makes all POST and PUT requests (except for POST requests with a
+ * very tiny request body) use the "Expect: 100-continue" header. This header
+ * allows the server to deny the operation early so that libcurl can bail out
+ * before having to send any data. This is useful in authentication
+ * cases and others.
+ *
+ * However, many servers don't implement the Expect: stuff properly and if the
+ * server doesn't respond (positively) within 1 second libcurl will continue
+ * and send off the data anyway.
+ *
+ * You can disable libcurl's use of the Expect: header the same way you disable
+ * any header, using -H / CURLOPT_HTTPHEADER, or by forcing it to use HTTP 1.0.
+ *
+ * This function is meant to be called after all headers from original request was called. It will
+ * append another header and set headers for a p_curl session
+ *
+ * @param p_curl reference to an easy curl session
+ * @param p_list list of headers as curl list
+ *
+ * @return az_result
```

Comment:
Do we want to list the set of az_results it can return and under what conditions? Similar to:
https://github.com/Azure/azure-sdk-for-c/blob/9c06eac1025670beb6cd4ccf0f0ae403e5f73dad/sdk/core/core/inc/az_span.h#L380-L385